### DISCUSSION TITLE
gpt keeps responding with same thing recently

### DISCUSSION THREAD

**Discussion by DatZig** [intitni/CopilotForXcode#214] [General]
hi there, not sure if I did something or what, but the past few days it seems no matter what I ask gpt it responds with the same things, telling me what the code I send it does. I highlight the code and use the command chat with selection. even if I type hi it responds with the same thing summarizing the code. what have I done? if I open a chat window with no selection and type hi it responds accordingly. I've tried restarting the agent/etc but can't seem to figure it out, guessing its something small... Thx for ur help!

**Comment by intitni** [comment]
I am not sure what is happening. It would be great if you can do MITM on the request sent to OpenAI to see what the prompts are.

When using the Chat with Selection command with selected code, it will update the system prompt to include the piece of code, which looks like:

```swift
return """
    \(language.isEmpty ? "" : "You must always reply in \(language)")
    You are a senior programmer, you will answer my questions concisely about the code below, or modify it according to my requests. When you receive a modification request, reply with the modified code in a code block.
    ```\(codeLanguage.rawValue)
    \(code)
    ```
    """
```

But it should still respond according to your message.
